clippy_lints/src/returns.rs

```diff
@@ -448,6 +455,7 @@ fn check_final_expr<'tcx>(
 
 fn emit_return_lint(
     cx: &LateContext<'_>,
+    lint_span: Span,
```

Comment:
Not super confident that this is the way we want to go - but it seemed like the lint_span and the suggestion_span had to diverge here. Happy to take suggestions!

Reply:
Ideally the suggestion span would be created inside this function, but that doesn't look like it will be trivial.